src/openforms/analytics_tools/models.py

```diff
@@ -0,0 +1,312 @@
+            "The XXXXX is your ID."
+        ),
+    )
+
+    enable_siteimprove_analytics = models.BooleanField(
+        _("enable siteImprove analytics"),
+        default=False,
+        help_text=_("Enabling this installs SiteImprove"),
+    )
+
+    analytics_cookie_consent_group = models.ForeignKey(
+        "cookie_consent.CookieGroup",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        help_text=_(
+            "The cookie group used for analytical cookies. The analytics scripts are "
+            "loaded only if this cookie group is accepted by the end-user."
+        ),
+    )
+
+    class Meta:
+        verbose_name = _("Analytics tools configuration")
+
+    def matomo_can_be_enabled(self) -> bool:
+        return (
+            self.matomo_url
+            and self.matomo_site_id
+            and self.enable_matomo_site_analytics
+            and self.analytics_cookie_consent_group
+        )
+
+    def piwik_can_be_enabled(self) -> bool:
+        return (
+            self.piwik_url
+            and self.piwik_site_id
+            and self.enable_piwik_site_analytics
+            and self.analytics_cookie_consent_group
+        )
+
+    def piwik_pro_can_be_enabled(self) -> bool:
+        return (
+            self.piwik_pro_url
+            and self.piwik_pro_site_id
+            and self.enable_piwik_pro_site_analytics
+            and self.analytics_cookie_consent_group
+        )
+
+    def siteimprove_can_be_enabled(self) -> bool:
+        return (
+            self.siteimprove_id
+            and self.enable_siteimprove_analytics
+            and self.analytics_cookie_consent_group
+        )
+
+    def google_analytics_can_be_enabled(self) -> bool:
+
+        return (
+            self.ga_code
+            and self.gtm_code
+            and self.enable_google_analytics
+            and self.analytics_cookie_consent_group
+        )
+
+    def update_analytical_cookies(self, cookies: dict, create: bool):
+        if create:
+            cookie_group = CookieGroup.objects.get(
+                id=self.analytics_cookie_consent_group.id
+            )
+            for cookie in cookies:
+                Cookie.objects.get_or_create(
+                    name=cookie["name"],
+                    path=cookie["path"],
+                    domain=settings.ALLOWED_HOSTS[0],
+                    cookiegroup=cookie_group,
+                )
+        else:
+            for cookie in cookies:
+                Cookie.objects.filter(name=cookie["name"]).delete()
+
+    def update_csp(self, csps: dict, create: bool):
+        if create:
+            for csp in csps:
+                CSPSetting.objects.get_or_create(
+                    directive=csp["directive"], value=csp["value"]
+                )
+        else:
+            for csp in csps:
+                CSPSetting.objects.filter(value=csp["value"]).delete()
+
+    def save(self, *args, **kwargs):
```

Comment:
django has a preferred order for methods within a model: https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#model-style